parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4

```diff
@@ -220,7 +220,21 @@ oidIndexClause
     ;
 
 createRelationalTableClause
-    : (LP_ relationalProperties RP_) collationClause? commitClause? physicalProperties? tableProperties?
+    : (LP_ relationalProperties RP_)
```

Comment:
Can you do some test for parse efficiency?

Reply:
Because there is a warning "*** contains an optional block with at least one alternative that can match an empty string."
I do this to resolve this warning.

Reply:
@strongduanmu

Reply:
This is old gramma of  partitionSpec:
```
	public final PartitionSpecContext partitionSpec() throws RecognitionException {
		PartitionSpecContext _localctx = new PartitionSpecContext(_ctx, getState());
		enterRule(_localctx, 502, RULE_partitionSpec);
		try {
			enterOuterAlt(_localctx, 1);
			{
			setState(5896);
			match(PARTITION);
			setState(5898);
			_errHandler.sync(this);
			switch ( getInterpreter().adaptivePredict(_input,626,_ctx) ) {
			case 1:
				{
				setState(5897);
				partitionName();
				}
				break;
			}
			setState(5901);
			_errHandler.sync(this);
			switch ( getInterpreter().adaptivePredict(_input,627,_ctx) ) {
			case 1:
				{
				setState(5900);
				tablePartitionDescription();
				}
				break;
			}
			}
		}
		catch (RecognitionException re) {
			_localctx.exception = re;
			_errHandler.reportError(this, re);
			_errHandler.recover(this, re);
		}
		finally {
			exitRule();
		}
		return _localctx;
	}
```

Reply:
This is new gramma of partitionSpec:
```
    public final OracleStatementParser.PartitionSpecContext partitionSpec() throws RecognitionException {
        OracleStatementParser.PartitionSpecContext _localctx = new OracleStatementParser.PartitionSpecContext(this._ctx, this.getState());
        this.enterRule(_localctx, 912, 456);

        try {
            this.enterOuterAlt(_localctx, 1);
            this.setState(9390);
            this.match(204);
            this.setState(9396);
            this._errHandler.sync(this);
            switch(((ParserATNSimulator)this.getInterpreter()).adaptivePredict(this._input, 1039, this._ctx)) {
            case 1:
                this.setState(9391);
                this.partitionName();
                this.setState(9392);
                this.tablePartitionDescription();
                break;
            case 2:
                this.setState(9394);
                this.partitionName();
                break;
            case 3:
                this.setState(9395);
                this.tablePartitionDescription();
            }
        } catch (RecognitionException var6) {
            _localctx.exception = var6;
            this._errHandler.reportError(this, var6);
            this._errHandler.recover(this, var6);
        } finally {
            this.exitRule();
        }

        return _localctx;
    }
```

Reply:
Only one switch sentence could cover all branches in the new gramma. In contrast, there are two switch sentences in the old gramma.

Reply:
Can you test with JMH? The new way of writing looks more complicated, we need to know how much performance can be improved by this way?